### PR TITLE
Rostests/fix kmtests

### PR DESCRIPTION
**Pull request by tkreuzer**
Fix some kmtests

Tests:
Build: https://build.reactos.org/#/builders/2/builds/2754
Test WHS: https://reactos.org/testman/compare.php?ids=85688,85716
Some of the tests that are fixed are not from this PR, but rather from a former fix by Thomas that was already merged.

### CODE REVIEW

File: modules/rostests/kmtests/ntos_mm/ZwAllocateVirtualMemory.c [hunk 1]
```diff
@@ -39,8 +39,8 @@ typedef struct _TEST_CONTEXT
         else if (!NT_SUCCESS(Status))                                                                                      \
             ok_eq_pointer(BaseAddress, NULL);                                                                              \
         RegionSize = 0;                                                                                                    \
-        Status = ZwFreeVirtualMemory(ProcessHandle, &BaseAddress, &RegionSize, MEM_RELEASE);                               \
-        if (FreeStatus != IGNORE) ok_eq_hex(Status, FreeStatus);                                                           \
+        Status2 = ZwFreeVirtualMemory(ProcessHandle, &BaseAddress, &RegionSize, MEM_RELEASE);                               \
+        if (FreeStatus != IGNORE) ok_eq_hex(Status2, FreeStatus);                                                           \
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply modules/rostests/kmtests/ntos_mm/ZwAllocateVirtualMemory.c]: Yes, it's to be able to distinguish the errors.

File: modules/rostests/kmtests/include/kmt_test.h [hunk 2]
```diff
@@ -233,7 +233,7 @@ VOID KmtFreeGuarded(PVOID Pointer);
 #ifdef KMT_KERNEL_MODE
 #define ok_irql(irql)                       ok(KeGetCurrentIrql() == irql, "IRQL is %d, expected %d\n", KeGetCurrentIrql(), irql)
 #endif /* defined KMT_KERNEL_MODE */
-#define ok_eq_print(value, expected, spec)  ok((value) == (expected), #value " = " spec ", expected " spec "\n", value, expected)
+#define ok_eq_print(value, expected, spec)  ok((value) == (expected), #value " = " spec ", expected " #expected "(" spec ")\n", value, expected)
```
Comment (learn-more) [inline modules/rostests/kmtests/include/kmt_test.h]: `spec` is now evaluated twice?

Comment (tkreuzer) [reply modules/rostests/kmtests/include/kmt_test.h]: No, it just adds the expected value as a string as it was passed to the macro. Well it was supposed to, but in fact it doesn't help much, as this macro is used from another macro, which already evaluates it. I'll try to improve it.

**Review by HBelusca — APPROVED** [review]

File: modules/rostests/kmtests/ntos_ex/ExHardError.c [hunk 3]
```diff
@@ -116,12 +117,12 @@ TestHardError(
...
     CheckHardError(0x40000013,                  0, OptionYesNoCancel,       STATUS_SUCCESS,            ResponseCancel,         0, 0);                          // outputs a box :|
     }
-    CheckHardError(0x40000009,                  0, 9,                       STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000a,                  0, 10,                      STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000b,                  0, 11,                      STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000c,                  0, 12,                      STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000d,                  0, MAXULONG / 2 + 1,        STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000d,                  0, MAXULONG,                STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
+    CheckHardError(0x40000009,                  0, 9,                       STATUS_SUCCESS,            ResponseReturnToCaller, 0, 0);
```
Comment (ThFabba) [inline modules/rostests/kmtests/ntos_ex/ExHardError.c:120]: I have no idea what's going on with these. My Win2003 returns `ResponseNotHandled` for these, and the logic of the code in csrsrv/winsrv makes it pretty clear that that's what we _should_ be getting here. I need to do some analysis of what's going on on the WHS bot to understand why it's returning `ResponseReturnToCaller`.

File: modules/rostests/kmtests/ntos_ke/KeEvent.c [hunk 4]
```diff
@@ -320,6 +320,12 @@ TestEventScheduling(
                "[%lu] Counter 0 = %lu\n",
                PriorityIncrement, ThreadData->CounterValues[0]);
         }
+        else if (PriorityIncrement >= 3)
```
[the remark this answers is not in the document]
Comment (HBelusca) [reply modules/rostests/kmtests/ntos_ke/KeEvent.c:323]: Could this be a single-vs.-multi processor effect?

File: modules/rostests/kmtests/ntos_ex/ExHardError.c (hunk 3 above)
Comment (HBelusca) [reply modules/rostests/kmtests/ntos_ex/ExHardError.c:120]: Could it be related to the non-initialized variable vulnerability in NtRaiseHardError (on Windows) that you mentioned on #kernel-dev on Mattermost?

Comment (ThFabba) [reply modules/rostests/kmtests/ntos_ex/ExHardError.c:120]: That one is triggered by the `OptionShutdownSystem` call above. These here should successfully make it all the way to winsrv and be failed there with `ResponseNotHandled`.

**Comment by HBelusca** [comment]
@tkreuzer What's the current status of this PR of yours?

**Pull request closed without merging**